pkg/apis/hive/v1/clusterdeployment_types.go

```diff
@@ -115,12 +106,28 @@ type ClusterDeploymentSpec struct {
 
 	// Installed is true if the cluster has been installed
 	Installed bool `json:"installed"`
+
+	// Provisioning contains settings used only for initial cluster provisioning.
+	// May be unset in the case of adopted clusters.
+	Provisioning *Provisioning `json:"provisioning"`
+}
+
+// Provisioning contains settings used only for initial cluster provisioning.
+type Provisioning struct {
+	// InstallConfigSecret is the reference to a secret that contains an openshift-install
+	// InstallConfig. This file will be passed through directly to the installer.
+	// Any version of InstallConfig can be used, provided it can be parsed by the openshift-install
+	// version for the release you are provisioning.
+	InstallConfigSecret corev1.LocalObjectReference `json:"installConfigSecret,omitempty"`
```

Comment:
This should not be `omitempty`. It is required when the `Provisioning` field is non-nil.